CMP.ServiceFabricReceiver.Stateful/EventProcessor.cs

```diff
@@ -6,23 +6,22 @@
 using System.Threading;
 using System.Threading.Tasks;
 using CMP.ServiceFabricReceiver.Common;
-using Microsoft.ServiceFabric.Data;
 
 namespace CMP.ServiceFabricReceiver.Stateful
 {
     public class EventProcessor : IEventProcessor
     {
-        private readonly Func<IDisposable> _operationLogger;
+        private readonly Func<IReadOnlyCollection<EventData>, string, Func<Task>, Task> _operationLogger;
         private readonly ILogger _logger;
         private readonly Action<string, object[]> _serviceEventSource;
-        private readonly Func<IReadOnlyCollection<EventData>, CancellationToken, Task> _handleEvents;
+        private readonly Func<string, Func<IReadOnlyCollection<EventData>, CancellationToken, Task>> _handleEvents;
         private readonly int exceptionDelaySeconds;
 
         public EventProcessor(
-            Func<IDisposable> operationLogger,
+            Func<IReadOnlyCollection<EventData>, string, Func<Task>, Task> operationLogger,
             ILogger logger,
             Action<string, object[]> serviceEventSource,
-            Func<IReadOnlyCollection<EventData>, CancellationToken, Task> handleEvents,
+            Func<string, Func<IReadOnlyCollection<EventData>, CancellationToken, Task>> handleEvents,
```

Comment:
It's hard to know what these string parameters is. Would be good to use either a named delegate or another more specific type.

Reply:
yeah, might be an improvement. Is kind of internal.
I also think the aspects of processing should be done by composition, meaning removing the "retry once" logic, to be able to build this trough composing aspects when registering the eventhandler.

Reply:
done